src/server/hook_func.c

```diff
@@ -5103,31 +5115,26 @@ sync_mom_hookfiles(void *minfo)
 
 		pbs_errno=0;
 		for (j=0; j < minfo_array[i]->mi_num_action; j++) {
+			hook *phook;
 			pact = minfo_array[i]->mi_action[j];
-
 			if ((pact == NULL) ||
 				(pact->action == MOM_HOOK_ACTION_NONE))
 				continue;
-
-			if (pact->action & MOM_HOOK_ACTION_DELETE) {
+			phook = find_hook(pact->hookname);
+			if (phook == NULL)
+				continue;
+			if (pact->action & MOM_HOOK_ACTION_DELETE)
 				action_expected++;
-			} else {
-				if (pact->action & MOM_HOOK_ACTION_SEND_ATTRS) {
-					action_expected++;
-				}
-				if (pact->action & MOM_HOOK_ACTION_SEND_SCRIPT) {
-					action_expected++;
-				}
-				if (pact->action & MOM_HOOK_ACTION_SEND_CONFIG) {
-					action_expected++;
-				}
-			}
-
-			if (pact->action & MOM_HOOK_ACTION_DELETE_RESCDEF) {
+			if (pact->action & MOM_HOOK_ACTION_SEND_ATTRS)
 				action_expected++;
-			} else if (pact->action & MOM_HOOK_ACTION_SEND_RESCDEF) {
+			if (pact->action & MOM_HOOK_ACTION_SEND_SCRIPT)
+				action_expected++;
+			if (pact->action & MOM_HOOK_ACTION_SEND_CONFIG)
+				action_expected++;
+			if (pact->action & MOM_HOOK_ACTION_DELETE_RESCDEF)
+				action_expected++;
+			else if (pact->action & MOM_HOOK_ACTION_SEND_RESCDEF)
 				action_expected++;
```

Comment:
Rather than testing each `MOM_HOOK_ACTION` flag and increment, why not count the number of set bits in `pact->action`. The code will be much shorter.